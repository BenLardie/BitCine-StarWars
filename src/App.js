import React from 'react';
import './App.css';
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Link
} from "react-router-dom";
import CharacterList from './Componets/CharacterList';



function App() {
  return (
      <Router>
        <Switch>
          <Route path="/">
            <CharacterList />
          </Route>
        </Switch>
      </Router>
  );
}

export default App;
