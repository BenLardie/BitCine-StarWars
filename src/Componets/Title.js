// eslint-disable-next-line no-unused-vars
import React from 'react'
/** @jsx jsx */
import { jsx } from '@emotion/core'

export default function Title(props) {
    return (
    <h1 css={{
            textAlign: 'center',
            color: 'white',
            fontFamily: "'Rubik Mono One', sans-serif",
        }}>
        {props.text}
    </h1>
    )
}
