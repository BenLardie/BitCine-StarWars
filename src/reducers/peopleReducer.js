import {
    FETCH_PEOPLE_BEGIN,
    FETCH_PEOPLE_SUCCESS,
    FETCH_PEOPLE_FAILURE
  } from "../actions/peopleActions";
  
  const initialState = {
    people: [],
    loading: false,
    url: 'https://swapi.co/api/people/',
    error: null
  };
  
  export default function productReducer(
    state = initialState,
    action
  ) {
    switch (action.type) {
      case FETCH_PEOPLE_BEGIN:
        // Mark the state as "loading" so we can show a spinner or something
        // Also, reset any errors. We're starting fresh.
        return {
          ...state,
          loading: true,
          error: null
        };
  
      case FETCH_PEOPLE_SUCCESS:
        // All done: set loading "false".
        // Also, replace the people with the ones from the server
        console.log(action)
        return {
          ...state,
          loading: false,
          people: [...state.people, ...action.payload.people.results],
          url: action.payload.people.next
        };
  
      case FETCH_PEOPLE_FAILURE:
        // The request failed, but it did stop, so set loading to "false".
        // Save the error, and we can display it somewhere
        // Since it failed, we don't have people to display anymore, so set it empty.
        // This is up to you and your app though: maybe you want to keep the people
        // around! Do whatever seems right.
        return {
          ...state,
          loading: false,
          error: action.payload.error,
          people: []
        };
  
      default:
        // ALWAYS have a default case in a reducer
        return state;
    }
  }
  