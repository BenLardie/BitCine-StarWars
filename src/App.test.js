import React from 'react';
import renderer from 'react-test-renderer'
import App from './App'
import { Provider } from 'react-redux'
import store from './store'
import Title from '../src/Components/Title'
import peopleReducer from './reducers/peopleReducer'


describe('Renders whole app with provider and compares it to a snapshot', () => {
  test('snapshot renders', ()=>{
    const component = renderer.create(
    <Provider store={store}>
      <App />
    </Provider>)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('Title Snapshot test', () => {
  test('snapshot renders', ()=> {
    const component = renderer.create(<Title text={'test'} />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})


describe('people reducer', () => {
  it('should return the intial state', () => {
    peopleReducer(undefined, {})
    expect.objectContaining({
      people: [],
      loading: false,
      url: 'https://swapi.co/api/people/',
      previous_url: 'https://swapi.co/api/people/?page=1', 
      error: null
    })
  })
})
