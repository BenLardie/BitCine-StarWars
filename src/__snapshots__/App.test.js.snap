// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders whole app with provider and compares it to a snapshot snapshot renders 1`] = `
<div
  className="css-17ykx3g-Home"
>
  <h1
    className="css-1iwaw1b-Title"
  >
    BitCine SWAPI Challenge
  </h1>
  <section
    className="css-5l9xtq-CharacterList"
  >
    <table
      className="css-1ktkhbs-CharacterList"
    >
      <tbody>
        <tr
          className="css-1l1d4v6-CharacterList"
        >
          <th />
          <th>
            NAME
          </th>
          <th>
            BIRTH YEAR
          </th>
          <th>
            HEIGHT
          </th>
          <th>
            MASS
          </th>
        </tr>
      </tbody>
    </table>
  </section>
</div>
`;

exports[`Title Snapshot test snapshot renders 1`] = `
<h1
  className="css-1iwaw1b-Title"
>
  test
</h1>
`;
